tests/python_package_test/test_sklearn.py

```diff
@@ -910,10 +938,8 @@ def test_metrics():
     assert 'multi_logloss' in gbm.evals_result_['training']
     assert 'multi_error' in gbm.evals_result_['training']
 
-    # invalid objective is replaced with default multiclass one
-    # and invalid binary metric is replaced with multiclass alternative
-    gbm = lgb.LGBMClassifier(objective='invalid_obj',
-                             **params).fit(eval_metric='binary_error', **params_fit)
```

Comment:
I believe this behavior is controversial and I'd better remove such silence replacement of objective function and instead raise error.

Reply:
I agree. We should explicitly tell the user what they intended to use is not valid. Or at least we should give a warning.